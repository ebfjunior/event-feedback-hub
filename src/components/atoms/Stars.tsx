import { Star } from 'lucide-react';
import { cn } from '@/lib/utils';

export function Stars({ rating, className }: { rating: number; className?: string }) {
  return (
    <div className={cn('flex items-center gap-1', className)} aria-label={`${rating} out of 5 stars`}>
      {Array.from({ length: 5 }).map((_, i) => (
        <Star key={i} className={cn('h-4 w-4', i < rating ? 'fill-yellow-400 stroke-yellow-400' : 'stroke-muted-foreground')} />
      ))}
    </div>
  );
}


