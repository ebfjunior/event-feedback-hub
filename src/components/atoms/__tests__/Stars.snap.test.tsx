import React from 'react';
import { render } from '@testing-library/react';
import { Stars } from '../Stars';

vi.mock('lucide-react', async () => {
	const actual = await vi.importActual<any>('lucide-react');
	return {
		...actual,
		Star: (props: any) => <svg data-icon="star" {...props} />,
	};
});

describe('Stars (snapshot)', () => {
	it('renders with rating', () => {
		const { container } = render(<Stars rating={3} />);
		expect(container.firstChild).toMatchSnapshot();
	});
});


