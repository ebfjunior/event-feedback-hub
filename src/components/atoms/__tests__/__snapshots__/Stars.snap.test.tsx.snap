// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Stars (snapshot) > renders with rating 1`] = `
<div
  aria-label="3 out of 5 stars"
  class="flex items-center gap-1"
>
  <svg
    class="h-4 w-4 fill-yellow-400 stroke-yellow-400"
    data-icon="star"
  />
  <svg
    class="h-4 w-4 fill-yellow-400 stroke-yellow-400"
    data-icon="star"
  />
  <svg
    class="h-4 w-4 fill-yellow-400 stroke-yellow-400"
    data-icon="star"
  />
  <svg
    class="h-4 w-4 stroke-muted-foreground"
    data-icon="star"
  />
  <svg
    class="h-4 w-4 stroke-muted-foreground"
    data-icon="star"
  />
</div>
`;
